### ISSUE TITLE
strange memory-leak like behaviour in 0.9.1

### ISSUE THREAD

**Issue by nergdron** [hashicorp/vault#3951]
We're seeing strange behaviour on our vault masters, where memory usage just grows over time until it uses all free RAM on the host, even though that usage is far more than the actual dataset in postgresql.

looking at the telemetry data, this specifically stands out:

```
'vault.runtime.alloc_bytes': 556320128.000
'vault.runtime.sys_bytes': 1414622464.000
```

the sys_bytes counter seems to consistently just rise, and never free to the OS, while the actual allocated bytes stays pretty much steady.

I wanted to test with the latest release (v0.9.3 as of now), but can't because we depend on Okta and issue #3944 is currently breaking things for us in anything > 0.9.1.

**Environment:**
* Vault Version: Vault v0.9.1 ('87b6919dea55da61d7cd444b2442cabb8ede8ab1'
* Operating System/Architecture: Ubuntu 16.04, on EC2

**Vault Config File:**
```
cluster_name = "dev-vault"

storage "postgresql" {
  connection_url = "postgres://some:creds@pg/vault"
}

ha_storage "etcd" {
  redirect_addr   = "https://172.23.187.65:8200"
  etcd_api   = "v3"
  ha_enabled = "true"
}

# listener for local connections on the vault host
listener "tcp" {
  tls_disable = 1
  address     = "127.0.0.1:8220"

  # duplicate config to ensure nothing tries to talk on 127.0.0.1
  cluster_address = "172.23.187.65:8201"
}

# listener for DNS RR shared address inside VPC
listener "tcp" {
  address         = "0.0.0.0:8200"
  cluster_address = "172.23.187.65:8201"
  tls_cert_file   = "/etc/letsencrypt/live/vault.domain/fullchain.pem"
  tls_key_file    = "/etc/letsencrypt/live/vault.domain/privkey.pem"
}
```

**Startup Log Output:**
```
Feb  8 19:40:00 ip-172-23-187-65 vault[12823]: 2018/02/08 19:40:00.938103 [INFO ] core: vault is sealed
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m==> Vault server configuration:
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m              HA Storage: etcd#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m                     Cgo: disabled#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m         Cluster Address: https://172.23.187.65:8201#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m              Listener 1: tcp (addr: "127.0.0.1:8220", cluster address: "172.23.187.65:8201", tls: "disabled")#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m              Listener 2: tcp (addr: "0.0.0.0:8200", cluster address: "172.23.187.65:8201", tls: "enabled")#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m               Log Level: info#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m                   Mlock: supported: true, enabled: true#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m        Redirect Address: https://172.23.187.65:8200#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m                 Storage: postgresql#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m                 Version: Vault v0.9.3#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m             Version Sha: 5acd6a21d5a69ab49d0f7c0bf540123a9b2c696d#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m#033[0m
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0;0m==> Vault server started! Log data will stream in below:
Feb  8 19:40:01 ip-172-23-187-65 vault[25468]: #033[0m
Feb  8 20:10:29 ip-172-23-187-65 vault[25468]: 2018/02/08 20:10:29.740644 [INFO ] core: vault is unsealed
Feb  8 20:10:29 ip-172-23-187-65 vault[25468]: 2018/02/08 20:10:29.741242 [INFO ] core: entering standby mode
```

**Expected Behavior:**
memory usage stays fairly steady over time.

**Actual Behavior:**
memory usage slowly climbs until it causes problems with the OS.

**Steps to Reproduce:**
run vault.

**Important Factoids:**

**References:**

**Comment by briankassouf** [comment]
This should be fixed in master by #3893

**Comment by briankassouf** [comment]
And should be in the 0.9.4 release, along with the okta fix!
